api/simqueue/resources/queue.py

```diff
@@ -151,47 +164,315 @@ async def get_log(
 
 @router.post("/jobs/", response_model=AcceptedJob, status_code=status_codes.HTTP_201_CREATED)
 async def create_job(
+    
+    
     job: SubmittedJob,
+    #tag: List[str] = Query(None, description="tags"),
+    #collab_id: List[str] = Query(None, description="collab id"),
+    #user_id: List[str] = Query(None, description="user id"),
+    #hardware_platform: List[str] = Query(None, description="hardware platform (e.g. SpiNNaker, BrainScales)"),
+    as_admin: bool = Query(False, description="Run this query with admin privileges, if you have them"),
     token: HTTPAuthorizationCredentials = Depends(auth)
 ):
-    pass
+    
+    get_user_task = asyncio.create_task(oauth.User.from_token(token.credentials))
+    user = await get_user_task
+    if (as_admin and user.is_admin)  or  user.can_edit(job.collab_id):
 
+        accepted_job = await db.post_job(user.username,job)
+        return accepted_job
+    raise HTTPException(
+        status_code=status_codes.HTTP_404_NOT_FOUND,
+        detail=f"You do not have access to this collab or there is no collab with this id"
+    )
 
 @router.put("/jobs/{job_id}", status_code=status_codes.HTTP_200_OK)
 async def update_job(
     job: JobPatch,
-    job_id: str = Path(..., title="Job ID", description="ID of the job to be retrieved"),
+    job_id: int = Path(..., title="Job ID", description="ID of the job to be retrieved"),
+    as_admin: bool = Query(False, description="Run this query with admin privileges, if you have them"),
     token: HTTPAuthorizationCredentials = Depends(auth)
 ):
     """
     For use by job handlers to update job metadata
     """
-    pass
+    
+    get_user_task = asyncio.create_task(oauth.User.from_token(token.credentials))
+    get_job_task = asyncio.create_task(db.get_job(job_id))
+    user = await get_user_task
+    oldJob = await get_job_task
```

Comment:
by convention Python variable names should be all lowercase, e.g. `old_job` rather than `oldJob`